### PR TITLE
Week 3 and 4 Exercises

### PR DESCRIPTION
**Pull request by Liliashamirian**

### CODE REVIEW

File: src/week3/Ex4.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        System.out.println("Array 1 : " + Arrays.toString(array1));
+        System.out.println("Array 2 : " + Arrays.toString(array2));
+        System.out.println("Combined Array Is : " + Arrays.toString(combineTwoArrays(array1, array2)));
+        System.out.println();
+        System.out.println("Sum of Two Arrays Average is : " + sumOfTwoArraysAverage(array1, array2));
+
+    }
+
+    private static int[] combineTwoArrays(int[] array1, int[] array2) {
+        int[] array3 = new int[array1.length + array2.length];
```
Comment (Smbat888) [inline src/week3/Ex4.java:30]: your solution is correct!

but plaese try to get familiar with the Arrays copyof method to combine two arrays))

File: src/week3/Ex5.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            array[i] = (int) (Math.random() * 10);
+        }
+        System.out.println("Initial Array : " + Arrays.toString(array));
+        System.out.println();
+        String strDouble = String.format("%.1f", increaseGivenElementByTenPercent(2, array));
+        System.out.print(strDouble + ", ");
+    }
+
+    private static double increaseGivenElementByTenPercent(int index, int[] array) {
+        double result = array[index] * 0.10;
```
Comment (Smbat888) [inline src/week3/Ex5.java]: unnecessary variable result:

just return array[index] * 0.10; instead

File: src/week4/StringAPI.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    }
+
+    //Ex2
+    public int getWordCount(String input) {
+        int count = 0;
+        if (input == null || input.isEmpty()) {
+            System.out.println(INVALID);
+            return 0;
+        }
+        if (!(input.matches("(.*) (.*)"))) {
```
Comment (Smbat888) [inline src/week4/StringAPI.java:27]: wow :)

File: src/week4/StringAPI.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    //Ex4
+    public String checkBrackets(String input) {
+        if (input == null || input.isEmpty()) {
+            return INVALID;
+        }
+        int length = input.length();
+        if (length % 2 != 0) {
+            return INVALID;
+        }
+        if (input.matches("[()]+")) {
```
Comment (Smbat888) [inline src/week4/StringAPI.java:64]: also you can do the following to covery other unexpected cases:

if(input.startsWith(")") || input.endsWith("(")) {
      return INVALID;
}

File: src/week4/StringAPI.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    }
+
+    //Ex8
+    public String removeDuplicateChars(String input) {
+        if (input == null || input.isEmpty()) {
+            return INVALID;
+        }
+        char[] arr1 = input.toCharArray();
+        String orgString = "";
+        for (char value : arr1) {
```
Comment (Smbat888) [inline src/week4/StringAPI.java:118]: esi senc ashxatela? mi hat kbacatres sa vonca ashxatum, PR-in el comment anes ok a)

File: src/week4/StringTest.java [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    public void checkPalindrome() {
+        String result;
+        result = api.checkIfInputIsPalindrome("");
+        assert result.equals(OUT_OF_BOUND);
+
+        result = api.checkIfInputIsPalindrome(null);
+        assert result.equals(OUT_OF_BOUND);
+
+        result = api.checkIfInputIsPalindrome("madam");
+        assert result.equals("Palindrome");
```
Comment (Smbat888) [inline src/week4/StringTest.java:68]: sranq el arjer constants sarqel

File: src/week4/StringTest.java [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+        result = api.checkIfInputIsPalindrome("Lilia");
+        assert result.equals("Not Palindrome");
+    }
+
+    //Ex4
+    public void checkBrackets() {
```
Comment (Smbat888) [inline src/week4/StringTest.java:75]: mi qani tex @Test annotationy moracel es)))

**Review by Smbat888 — COMMENTED** [review]
apres!, asacners uxxi, bayc menak mi hat ban kasem,
ete mutq es anum incvor ban error handlingy ampayman ara..
orinak asum es scanner.nextInt() .... usery kara character mutq ani u qo mot exception qci, programt chashxati, petqa handle anes vat caser-y)

File: src/week4/StringTest.java (hunk 7 above)
Comment (Liliashamirian) [reply src/week4/StringTest.java:75]: haaa :D 
Im mot testng exceptiona kcum u chem karoghacel test anel : ))
kavelacnem

File: src/week4/StringTest.java (hunk 6 above)
Comment (Liliashamirian) [reply src/week4/StringTest.java:68]: ok

File: src/week4/StringAPI.java (hunk 5 above)
Comment (Liliashamirian) [reply src/week4/StringAPI.java:118]: ha ashkhatumer
mi hat original string uni vor skzbum empty a
for ov fruma u amen array i index i vra kanga arnum stuguma ed original stringum eghac letternere gtnuma te che, yete chi gtnum -1 a veradarcnum u ed mi hat elementy avelacnuma original stringin, hajort ankam arten hasnuma 2rord elementin u eli mana galis originalum eghac letternere edegh kan te che, yete -1 chveradarcni ed letter e chi avelacni

File: src/week4/StringAPI.java (hunk 4 above)
Comment (Liliashamirian) [reply src/week4/StringAPI.java:64]: ok 👍

File: src/week4/StringAPI.java (hunk 3 above)
Comment (Liliashamirian) [reply src/week4/StringAPI.java:27]: :)

File: src/week3/Ex5.java (hunk 2 above)
Comment (Liliashamirian) [reply src/week3/Ex5.java]: ok

File: src/week3/Ex4.java (hunk 1 above)
Comment (Liliashamirian) [reply src/week3/Ex4.java:30]: ok thanks

**Pull request closed without merging**